index.js

```diff
@@ -166,66 +168,130 @@ function provider ({ Biome, blocks, blocksByStateId, toolMultipliers, shapes, ma
     return heldItemType && this.harvestTools && this.harvestTools[heldItemType]
   }
 
-  function effectLevel (effect, effects) {
-    const e = effects[effectsByName[effect].id]
+  // Determine data required to actually compute dig times that is version-dependent
+  let blockDigTimeData
+
+  // 1.17+: effect names have been fixed to actually match their registry names
+  if (protocolVersion >= 755) {
+    blockDigTimeData = {
+      enchantmentKey: 'efficiency',
+      hasteEffectName: 'haste',
+      miningFatigueEffectName: 'mining_fatigue',
+      conduitPowerEffectName: 'conduit_power',
+      aquaAffinityEnchantmentName: 'aqua_affinity'
+    }
+  } else {
+    // 1.13+: enchantments are no longer stored by IDs, but rather by registry names
+    const isPost113 = protocolVersion >= 393
+    blockDigTimeData = {
+      enchantmentKey: isPost113 ? 'efficiency' : 32,
+      hasteEffectName: 'Haste',
+      miningFatigueEffectName: 'MiningFatigue',
+      conduitPowerEffectName: 'ConduitPower',
+      aquaAffinityEnchantmentName: 'AquaAffinity'
+    }
+  }
+
+  function getEffectLevel (effectName, effects) {
+    const e = effects[effectsByName[effectName].id]
     return e ? e.amplifier : -1
   }
 
-  function enchantmentLevel (enchantment, enchantments) {
+  function getEnchantmentLevel (enchantmentKey, enchantments) {
     for (const e of enchantments) {
-      if (typeof enchantment === 'string' ? e.id.includes(enchantment) : e.id === enchantment) {
+      if (typeof enchantmentKey === 'string' ? e.id.includes(enchantmentKey) : e.id === enchantmentKey) {
         return e.lvl
       }
     }
     return -1
   }
 
-  // http://minecraft.gamepedia.com/Breaking#Calculation
+  function getMiningFatigueMultiplier (effectLevel) {
+    switch (effectLevel) {
+      case 0: return 0.0
+      case 1: return 0.3
+      case 2: return 0.09
+      case 3: return 0.0027
+      default: return 8.1E-4
+    }
+  }
+
+  // Reworked to completely match Minecraft 1.7 block breaking code
+  // This has not introduced any considerable changes though, so for older version behaviour should still remain consistent
```

Comment:
not really needed to say this, this is expected if the pr is merged

Reply:
fair point